component/src/main/java/io/siddhi/extension/store/elasticsearch/ElasticsearchConfigs.java

```diff
@@ -0,0 +1,532 @@
+import static io.siddhi.extension.store.elasticsearch.utils.ElasticsearchTableConstants.DEFAULT_NUMBER_OF_SHARDS;
+import static io.siddhi.extension.store.elasticsearch.utils.ElasticsearchTableConstants.DEFAULT_PASSWORD;
+import static io.siddhi.extension.store.elasticsearch.utils.ElasticsearchTableConstants.DEFAULT_PAYLOAD_INDEX_OF_INDEX_NAME;
+import static io.siddhi.extension.store.elasticsearch.utils.ElasticsearchTableConstants.DEFAULT_PORT;
+import static io.siddhi.extension.store.elasticsearch.utils.ElasticsearchTableConstants.DEFAULT_SCHEME;
+import static io.siddhi.extension.store.elasticsearch.utils.ElasticsearchTableConstants.DEFAULT_SSL_ENABLED;
+import static io.siddhi.extension.store.elasticsearch.utils.ElasticsearchTableConstants.DEFAULT_TRUSTSTORE_PASS;
+import static io.siddhi.extension.store.elasticsearch.utils.ElasticsearchTableConstants.DEFAULT_TRUSTSTORE_TYPE;
+import static io.siddhi.extension.store.elasticsearch.utils.ElasticsearchTableConstants.DEFAULT_USER_NAME;
+
+/**
+ * Configuration for the elasticsearch sink and the store.
+ */
+public class ElasticsearchConfigs {
+
+    private static final Logger logger = Logger.getLogger(ElasticsearchConfigs.class);
+    private RestHighLevelClient restHighLevelClient;
+    private List<Attribute> attributes;
+    private String hostname = DEFAULT_HOSTNAME;
+    private String indexName;
+    private String indexAlias;
+    private int port = DEFAULT_PORT;
+    private String scheme = DEFAULT_SCHEME;
+    private String userName = DEFAULT_USER_NAME;
+    private String password = DEFAULT_PASSWORD;
+    private int numberOfShards = DEFAULT_NUMBER_OF_SHARDS;
+    private int numberOfReplicas = DEFAULT_NUMBER_OF_REPLICAS;
+    private BulkProcessor bulkProcessor;
+    private int bulkActions = DEFAULT_BULK_ACTIONS;
+    private long bulkSize = DEFAULT_BULK_SIZE_IN_MB;
+    private int concurrentRequests = DEFAULT_CONCURRENT_REQUESTS;
+    private long flushInterval = DEFAULT_FLUSH_INTERVAL;
+    private String backoffPolicy = DEFAULT_BACKOFF_POLICY;
+    private int backoffPolicyRetryNo = DEFAULT_BACKOFF_POLICY_RETRY_NO;
+    private long backoffPolicyWaitTime = DEFAULT_BACKOFF_POLICY_WAIT_TIME;
+    private int ioThreadCount = DEFAULT_IO_THREAD_COUNT;
+    private String trustStorePass = DEFAULT_TRUSTSTORE_PASS;
+    private String trustStorePath;
+    private String trustStoreType = DEFAULT_TRUSTSTORE_TYPE;
+    private boolean sslEnabled = DEFAULT_SSL_ENABLED;
+    private int payloadIndexOfIndexName = DEFAULT_PAYLOAD_INDEX_OF_INDEX_NAME;
+    private String listOfHostnames;
+    private String definitionId;
+    private Annotation storeAnnotation;
+    private ElasticsearchSink elasticsearchSink;
+
+    public ElasticsearchConfigs() { // calls from ElasticSearchEventTable
+
+    }
+
+    public ElasticsearchConfigs(ElasticsearchSink elasticsearchSink) { // calls from ElasticSearchSink
```

Comment:
Do we need both constructor and init method ?

Reply:
Yes, there are two constructors in the ElasticsearchConfig, so it's better to have both init and constructor separately.